### ISSUE TITLE
Track uses through first-class modules

### ISSUE THREAD

**Issue by fantazio** [LexiFi/dead_code_analyzer#83]
Analyzing code with first-class modules is untested and undocumented.

They are used in [Opam 2.5.1](https://github.com/ocaml/opam/tree/2.5.1) and involved in the following false positives in the `UNUSED CONSTRUCTORS/RECORD FIELDS` section:
```
/tmp/proj/opam/src/solver/opamCudfSolverSig.ml:12: criteria_def.crit_default
/tmp/proj/opam/src/solver/opamCudfSolverSig.ml:13: criteria_def.crit_upgrade
/tmp/proj/opam/src/solver/opamCudfSolverSig.ml:14: criteria_def.crit_fixup
/tmp/proj/opam/src/solver/opamCudfSolverSig.ml:15: criteria_def.crit_best_effort_prefix
```
The fields listed are used in the [`OpamSolverConfig.with_auto_criteria`](https://github.com/ocaml/opam/blob/2.5.1/src/solver/opamSolverConfig.ml#L158)

**Comment by fantazio** [comment]
Following discussion in https://github.com/LexiFi/dead_code_analyzer/issues/85, [goblint/analyzer](https://github.com/goblint/analyzer) uses first-class modules a lot.
In `src`, the following patterns (typical of first-class modules) appear:
- `(module ` 739 times
- `let module` 190 times
- `(val ` 57 times

I have not triaged the results of the dead_code_analyzer on that project but, at first glance, there are false positives directly connected to this feature.
The project also uses lots of functors and module inclusions, which will require more robust testing.
